app/gui/src/controller/text.rs

```diff
@@ -96,19 +97,30 @@ impl Handle {
         }
     }
 
-    /// Store the given content to file.
+    /// Save the project to the VCS.
     #[profile(Detail)]
-    pub fn store_content(&self, content: String) -> impl Future<Output = FallibleResult> {
-        let file_handle = self.file.clone_ref();
+    pub fn save_project_to_vcs(&self) -> impl Future<Output = FallibleResult> {
+        let language_server = match self.file.clone_ref() {
+            FileHandle::PlainText { language_server, .. } => language_server,
+            FileHandle::Module { controller } => controller.language_server,
+        };
+        let project_root = language_server.project_root().id();
+        let path_segments: [&str; 0] = [];
+        let root_path = language_server::Path::new(project_root, &path_segments);
+
         async move {
-            match file_handle {
-                FileHandle::PlainText { path, language_server } =>
-                    language_server.write_file(&path, &content).await?,
-                FileHandle::Module { controller } => {
-                    controller.check_code_sync(content)?;
-                    controller.save_file().await?
+            let response = language_server.write_vcs(&root_path, &None).await;
+            if let Err(RpcError::RemoteError(json_rpc::messages::Error {
+                code: error_code, ..
+            })) = response
+            {
+                if error_code == code::FILE_NOT_FOUND {
+                    language_server.init_vcs(&root_path).await?;
+                    language_server.write_vcs(&root_path, &None).await?;
+                    return Ok(());
                 }
             }
+            response?;
```

Comment:
1. This method should be a part of the Project controller, as we no longer saving a single text file.
2. Please add a test for this method, mocking the communication with the engine. See other similar tests for reference.